framework/src/meshgenerators/SidesetsEncloseBlocks.C

```diff
@@ -0,0 +1,156 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SidesetsEncloseBlocks.h"
+#include "InputParameters.h"
+#include "MooseTypes.h"
+#include "MooseMeshUtils.h"
+#include "CastUniquePointer.h"
+
+#include "libmesh/remote_elem.h"
+
+registerMooseObject("MooseApp", SidesetsEncloseBlocks);
+
+InputParameters
+SidesetsEncloseBlocks::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addParam<std::vector<SubdomainName>>(
+      "block", "The set of blocks that are checked if they are enclosed by boundary");
+  params.addParam<std::vector<BoundaryName>>("boundary",
+                                             "The name of the boundaries enclosing the ");
+  params.addParam<BoundaryName>("new_boundary", "The name of the boundary to create");
+  params.addParam<bool>(
+      "allow_sideset_on_neighbor",
+      false,
+      "If true the mesh generator searches the neighbor side for a sideset as well");
+  params.addClassDescription(
+      "MeshGenerator that checks if a set of blocks is enclosed by a set of sidesets."
+      "It can either add sides that are not covered by a sideset by a new sidesets or"
+      "error out.");
+
+  return params;
+}
+
+SidesetsEncloseBlocks::SidesetsEncloseBlocks(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _neighbor_sideset(getParam<bool>("allow_sideset_on_neighbor"))
+{
+}
+
+std::unique_ptr<MeshBase>
+SidesetsEncloseBlocks::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+
+  BoundaryInfo & boundary_info = mesh->get_boundary_info();
+
+  // error on finding a side that is not covered
+  bool error_out = !isParamValid("new_boundary");
+  boundary_id_type new_sideset_id;
+  if (!error_out)
+  {
+    BoundaryName new_boundary = getParam<BoundaryName>("new_boundary");
+    std::stringstream ss;
+    ss << new_boundary;
+    ss >> new_sideset_id;
+    if (ss.fail())
+      new_sideset_id = boundary_info.get_id_by_name(new_boundary);
+
+    if (new_sideset_id == BoundaryInfo::invalid_id)
+      paramError("new_boundary", "Not a valid boundary");
+  }
+
+  std::vector<subdomain_id_type> vec_block_ids =
+      MooseMeshUtils::getSubdomainIDs(*mesh, getParam<std::vector<SubdomainName>>("block"));
+  std::set<subdomain_id_type> blk_ids(vec_block_ids.begin(), vec_block_ids.end());
+
+  // get boundaries
+  // check if the provided sideset name is actually a sideset id
+  // if it can be converted to integer it's interpreted
+  // as a sideset id
```

Comment:
This check (and the one above) seems like it's a fragile version of what's already in getBoundaryIDs; should we be using that instead?